Tests/ScipioKitTests/CacheSystemTests.swift

```diff
@@ -45,38 +36,112 @@ InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault
+            "sdks" : [
+              "iOS"
+            ]
+          },
+          "clangVersion" : "clang-1400.0.29.102",
+          "pin" : {
+            "revision" : "111111111"
+          },
+          "targetName" : "MyTarget",
+          "xcodeVersion" : {
+            "xcodeBuildVersion" : "15F31d",
+            "xcodeVersion" : "15.4"
+          }
+        }
+        """
+        // swiftlint:enable line_length
         XCTAssertEqual(rawString, expected)
     }
+
+    #if compiler(>=6.0)
+
+    func testCacheKeyCalculationForRootPackageTarget() async throws {
+        let fileSystem = localFileSystem
+        let testingPackagePath = fixturePath.appendingPathComponent("TestingPackage")
+        let tempTestingPackagePath = testingPackagePath.absolutePath.parentDirectory.appending(component: "temp_TestingPackage")
+
+        try fileSystem.removeFileTree(tempTestingPackagePath)
+        try fileSystem.copy(from: testingPackagePath.absolutePath, to: tempTestingPackagePath)
+
+        defer { try? fileSystem.removeFileTree(tempTestingPackagePath) }
+
+        let descriptionPackage = try DescriptionPackage(
+            packageDirectory: tempTestingPackagePath,
+            mode: .createPackage,
+            onlyUseVersionsFromResolvedFile: false
+        )
+        let cacheSystem = CacheSystem(
+            pinsStore: try descriptionPackage.workspace.pinsStore.load(),
+            outputDirectory: FileManager.default.temporaryDirectory.appendingPathComponent("XCFrameworks"),
+            storage: nil
+        )
+        let testingPackage = descriptionPackage
+            .graph
+            .packages
+            .first { $0.manifest.displayName == descriptionPackage.manifest.displayName }!
+
+        let myTarget = testingPackage.modules.first { $0.name == "MyTarget" }!
+        let cacheTarget = CacheSystem.CacheTarget(
+            buildProduct: BuildProduct(
+                package: testingPackage,
+                target: myTarget
+            ),
+            buildOptions: BuildOptions(
+                buildConfiguration: .release,
+                isDebugSymbolsEmbedded: false,
+                frameworkType: .dynamic,
+                sdks: [.iOS, .iOSSimulator],
+                extraFlags: nil,
+                extraBuildParameters: nil,
+                enableLibraryEvolution: false,
+                customFrameworkModuleMapContents: nil
+            )
+        )
+
+        // Ensure that the cache key cannot be calculated if the package is not in the Git repository.
+        do {
+            _ = try await cacheSystem.calculateCacheKey(of: cacheTarget)
+            XCTFail("A cache key should not be possible to calculate if the package is not in a repository.")
+        } catch let error as CacheSystem.Error {
+            XCTAssertEqual(error.errorDescription, "Repository version is not detected for \(descriptionPackage.name).")
+        } catch {
+            XCTFail("Wrong error type.")
+        }
```

Comment:
It's better to use `XCTAssertThrowErrors` instead
(Sorry for the other code uses this)

Reply:
Since `calculateCacheKe`y is an async function and there is no version of `XCTAssertThrowErrors` that supports concurrency, using the latter with the former becomes problematic. Let me know if you still see how it can be applied to this context.

Reply:
Make sense! Sorry for my wrong point